source/Tests/Testing/Integration/IntegrationTestContainerAnalyzerFixture.cs

```diff
@@ -204,28 +204,107 @@ public class TestClass : IntegrationTest
+
+    // public = not ok
+    public static string {|#5:SomePublicReadOnlyStringProp|} {get;} = ""foo"";
+
+    // ----- Methods are only OK if they are nonpublic -----
+
+    public static void {|#99:SomePublicMethod|}()  { }
+
+    static void SomePrivateMethod() { }
+
+    // ----- Nested types are all OK -----
+
+    public class SomeIntegrationTest : IntegrationTest  // good, class is an integration test
+    { }
+
+    class SomeClass{ }
+
+    struct SomeStruct{ }
+
+    enum SomeEnum{ }
+
+    // ----- fields holding collections -----
+
+    static readonly IEnumerable<string> SomeReadonlyEnumerableOfString = new[]{ ""foo"", ""bar"" };  // good, property is immutable and nonpublic
+    static readonly IReadOnlyList<string> SomeReadOnlyListOfString = new[]{ ""foo"", ""bar"" };  // good, property is immutable and nonpublic
+    static readonly IReadOnlyDictionary<int, string> SomeReadOnlyDict = new Dictionary<int, string>(){ [0] = ""foo"", [1] = ""bar"" };  // good, property is immutable and nonpublic
+    // static readonly IReadOnlySet<string> SomeReadOnlySet = new HashSet<string>(){ ""foo"", ""bar"" };  // good, property is immutable and nonpublic. COMMENTED: SEE NOTE
+    
+    static IEnumerable<string> {|#30:SomeMutableEnumerableOfString|} = new[]{ ""foo"", ""bar"" };  // bad, property is mutable
+
+    static readonly string[] {|#31:SomeArrayOfString|} = new[]{ ""foo"", ""bar"" };  // bad, property is readonly but array itself is mutable
+    static readonly List<string> {|#32:SomeListOfString|} = new(){ ""foo"", ""bar"" };  // bad, property is readonly but list itself is mutable
+    static readonly Dictionary<int, string> {|#33:SomeMutableDictionary|} = new Dictionary<int, string>(){ [0] = ""foo"", [1] = ""bar"" }; // bad, property is readonly but list itself is mutable
+    static readonly HashSet<string> {|#34:SomeMutableSet|} = new HashSet<string>(){ ""foo"", ""bar"" }; // bad, property is readonly but list itself is mutable
+
+    // properties holding collections not explicitly tested because it's the same logic as fields
 }
 ";
-            var propertyResult = new DiagnosticResult(Descriptors.Oct2006IntegrationTestContainersMustOnlyContainTypesAndMethods).WithLocation(0);
-            var fieldResult = new DiagnosticResult(Descriptors.Oct2006IntegrationTestContainersMustOnlyContainTypesAndMethods).WithLocation(1);
-            var methodResult = new DiagnosticResult(Descriptors.Oct2007IntegrationTestContainersMethodsMustBePrivate).WithLocation(2);
-            await Verify.VerifyAnalyzerAsync(container.WithTestingTypes(), propertyResult, fieldResult, methodResult);
+            // IReadOnlySet Note: Our DLL targets netstandard2.0, with the comment "As at Feb 2021 anything later than netstandard 2.0 doesn't work in Visual Studio".
+            // IReadOnlySet was added in .NET 5 so isn't in netstandard2.0, so, while it works in the real world, we can't hit it with unit tests
```

Comment:
🤔 We should be able to do this from the Unit Tests. They should target .net 6 since that's what the analysers will be run under.

The netstandard2.0 limitation on the analyzer dll should be kept (we just ran into that again recently).

Reply:
Yeah this one was 🤔 

The tests already target dotnet 6, but when I attempted to use the IReadOnlySet interface in the sourcecode, it didn't compile due to the type not existing.
I assumed this was down to the netstandard2.0 limitation and left it there.

Maybe we can revisit in future?